scripts/src/functions.bash

```diff
@@ -114,16 +120,98 @@ create-ns-and-developer-role-bindings() {
+  elif echo "${CI_COMMIT_REF_SLUG}" | grep -qE "^(${NON_DEVELOP_BRANCHES_REGEX})$"; then
+    echo "Your branch '${CI_COMMIT_REF_SLUG} matches the '${NON_DEVELOP_BRANCHES_REGEX}' regex, it means that we handle it as a production branch."
+    echo "The DEVELOP_GITLAB_AGENT_PROJECT and DEVELOP_GITLAB_AGENT_ID variables will be used, if present."
+  elif [ "${CI_SERVER_VERSION_MAJOR}" -ge "17" ]; then
+    echo "The GitLab Agent is not configured correctly. Please check the variables and the workflow."
+  else
+    echo "The GitLab Agent is not configured. We use the environment based cluster configuration."
+  fi
+}
+
+_setup-gitlab-agent-kubernetes-context() {
+  if [ "${1:-}" = "" ]; then
+    echo "Missing the GitLab Agent host project name."
+    exit 1
+  fi
+  if [ "${2:-}" = "" ]; then
+    echo "Missing the GitLab Agent ID."
+    exit 1
+  fi
+
+  echo "Switching Kubernetes context to use the context provided by the GitLab Agent."
+  echo "The used GitLab Agent ID is: ${2}"
+
+  kubectl config use-context "${1}:${2}"
+}
+
 setup-gitlab-agent() {
-  if [ -n "${GITLAB_AGENT_PROJECT:-}" ] && [ -n "${GITLAB_AGENT_ID:-}" ] && [ "${DISABLE_GITLAB_AGENT:-0}" != "1" ]; then
-    echo "The deployment will use the GitLab Agent."
-    echo "Switching Kubernetes context to use the context provided by the GitLab Agent."
-    kubectl config use-context "${GITLAB_AGENT_PROJECT}:${GITLAB_AGENT_ID}"
+  print-banner "SETUP GITLAB AGENT"
+  _gitlab-agent-print-vars
+  _gitlab-agent-print-workflow
+
+  # If the GitLab Agent is disabled, return early.
+  if [ "${DISABLE_GITLAB_AGENT:-0}" != "1" ]; then
+    return
+  fi
+
+  # If the GitLab Agent variables are configured to a specific project and ID, use them.
+  if [ -n "${GITLAB_AGENT_PROJECT:-}" ] && [ -n "${GITLAB_AGENT_ID:-}" ]; then
+    _setup-gitlab-agent-kubernetes-context "${GITLAB_AGENT_PROJECT}" "${GITLAB_AGENT_ID}"
+    return
+  fi
+
+  # If the current branch is non-production and the development variables are set, use the develop GitLab Agent.
+  # Please note the `-v` in the grep command that is used to invert the match.
+  if echo "${CI_COMMIT_REF_SLUG}" | grep -qvE "^(${NON_DEVELOP_BRANCHES_REGEX})$"; then
+    if [ -z "${DEVELOP_GITLAB_AGENT_PROJECT:-}" ] && [ -z "${DEVELOP_GITLAB_AGENT_ID:-}" ]; then
+      return
+    fi
+
+    _setup-gitlab-agent-kubernetes-context "${DEVELOP_GITLAB_AGENT_PROJECT}" "${DEVELOP_GITLAB_AGENT_ID}"
+    return
+  fi
+
+  # If the current branch is a production ones and the production variables are set, use the production GitLab Agent.
+  if echo "${CI_COMMIT_REF_SLUG}" | grep -qE "^(${NON_DEVELOP_BRANCHES_REGEX})$"; then
+    if [ -n "${PRODUCTION_GITLAB_AGENT_PROJECT:-}" ] && [ -n "${PRODUCTION_GITLAB_AGENT_ID:-}" ]; then
+      return
+    fi
+
+    _setup-gitlab-agent-kubernetes-context "${PRODUCTION_GITLAB_AGENT_PROJECT}" "${PRODUCTION_GITLAB_AGENT_ID}"
+    return
```

Comment:
I would add a new function to print that the agent is not configured.

```bash

_gitlab_agent_exit_not_configured() {
    echo "Gitlab agent not configured"
    // pseudocode to print a custom message if present.
    if arg(1)
      echo ": arg(1)"
    return 
}
```

Reply:
> I would add a new function to print that the agent is not configured.
> 
> ```shell
> _gitlab_agent_exit_not_configured() {
>     echo "Gitlab agent not configured"
>     // pseudocode to print a custom message if present.
>     if arg(1)
>       echo ": arg(1)"
>     return 
> }
> ```

The message for non-configured-agent is printed here: https://github.com/sparkfabrik/spark-k8s-deployer/pull/174/files#diff-96524718f63a467ac7a74e21082c918aa6e57208e1d794964a8c88fbb65f193cR159

Is this not enough?

Reply:
> Altra cosa che farei è un end banner, così da avere le sezioni ben separate.

Usata la stessa funzione `print-banner` con l'end della sezione.